inspirehep/modules/citations/receivers.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2015, 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+
+from invenio_records.signals import after_record_insert, after_record_update, before_record_index
+
+from .tasks import update_citation_count_for_records
+from .utils import calculate_citation_count
+
+
+@after_record_insert.connect
+def catch_citations_insert(sender, *args, **kwargs):
+    """Triggers update of citation count for all the records
+    which are affected by insertion of the sender
+
+    :param sender: The json representation of the record that is going to be inserted
+    """
+    records_to_notify = []
+    if 'references' in sender:
+            records_to_notify = [ref['recid'] for ref in sender['references'] if ref.get('recid')]
+
+    if records_to_notify:
+        # countdown added as it takes a while before an indexed record can be accessed from es
+        update_citation_count_for_records.apply_async(args=[records_to_notify], countdown=3)
+
+
+@before_record_index.connect
+def add_citation_count_on_insert_or_update(sender, *args, **kwargs):
+
+    if kwargs is not None:
+        json = kwargs.get('json')
+        index = kwargs.get('index')
+        if json and index == 'hep':
+            calculate_citation_count(sender, json)
+
+
+@after_record_update.connect
+def catch_citations_update(sender, *args, **kwargs):
+    """Triggers update of citation count for all the records
+    which are affected by update of the sender
+
+    :param sender: The json representation of the record that is going to be updated
+    """
+    refs_before_update = []
+    if sender.init_json:
```

Comment:
I am now realizing that `.init_json` sounds more like a method than a properties. Can you replace it with e.g. `.original_json`?